packages/api-key-stamper/src/utils.ts

```diff
@@ -23,17 +56,29 @@ function hexStringToUint8Array(input: string): Uint8Array {
     throw new Error(`Invalid hex string: ${JSON.stringify(input)}`);
   }
 
-  return Uint8Array.from(
+  const buffer = Uint8Array.from(
     input
       .match(
         /.{2}/g // Split string by every two characters
       )!
       .map((byte) => parseInt(byte, 16))
   );
+
+  if (!length) {
+    return buffer;
+  }
+
+  // If a length is specified, ensure we sufficiently pad
+  let paddedBuffer = new Uint8Array(length);
+  paddedBuffer.set(buffer, length - buffer.length);
+  return paddedBuffer;
```

Comment:
I think it's important to handle cases where `length` is _less_ than `buffer.length`. In this case, maybe an error?